### PR TITLE
Define nns-governance canister

### PR DESCRIPTION
**Pull request by bitdivine**
# Motivation
When creating an SNS, it is necessary to make calls to the nns-governance, and to do that in the normal way it is necessary to have the canister defined in dfx.json.

# Changes
- Add an entry for the governance canister in dfx.json

# Tests
- [x] Verify that the build configuration is unchanged
  - Checked mainnet config with:
    ```
     2036  git stash
     2037  DFX_NETWORK=mainnet ./config.sh
     2038  cp deployment-config.json ,before
     2039  git stash pop
     2040  meld <(jq --sort-keys . <,before) <(DFX_NETWORK=mainnet ./config.sh)
    ```
  - Ditto for the testnet

### CODE REVIEW

File: dfx.json [hunk 1]
```diff
@@ -1,6 +1,23 @@
...
+      "build": "./e2e-tests/scripts/nns-canister-download && ./e2e-tests/scripts/nns-canister-build",
+      "remote": {
+        "id": {
+          "local": "rrkah-fqaaa-aaaaa-aaaaq-cai",
+          "mainnet": "rrkah-fqaaa-aaaaa-aaaaq-cai",
+          "testnet": "rrkah-fqaaa-aaaaa-aaaaq-cai",
+          "small06": "rrkah-fqaaa-aaaaa-aaaaq-cai",
+          "small11": "rrkah-fqaaa-aaaaa-aaaaq-cai",
+          "small12": "rrkah-fqaaa-aaaaa-aaaaq-cai",
+          "staging": "rrkah-fqaaa-aaaaa-aaaaq-cai"
```
Comment (bitdivine) [inline dfx.json:17]: Note: It would be useful to be able to define a default!

File: .github/workflows/checks.yml [hunk 2]
```diff
@@ -124,6 +124,9 @@ jobs:
       - name: Checkout
         uses: actions/checkout@v2
```
Comment (bitdivine) [inline .github/workflows/checks.yml:125]: These tests run `./config.sh` which in turn now calls `dfx` to get the governance canister ID

File: config.sh [hunk 3]
```diff
@@ -50,10 +50,19 @@ local_deployment_data="$(
   : "Note: If you want to use a wasm canister deployed by someone else, add the canister ID to the remote section in dfx.json:"
   : "      dfx.json -> canisters -> wasm_canister -> remote -> id -> your DFX_NETWORK -> THE_WASM_CANISTER_ID"
   WASM_CANISTER_ID="$(dfx canister --network "$DFX_NETWORK" id wasm_canister 2>/dev/null || echo "NO_SNS_WASM_CANISTER_SPECIFIED")"
+  : "Get the governance canister ID - it should be defined"
+  GOVERNANCE_CANISTER_ID="$(dfx canister --network "$DFX_NETWORK" id nns-governance)"
+
   export WASM_CANISTER_ID
+  export GOVERNANCE_CANISTER_ID
 
   : "Put any values we found in JSON.  Omit any that are undefined."
```
Comment (bitdivine) [inline config.sh:59]: Below is mostly a formatting change - putting every value on a dedicated line.  New is the governance canister ID.

File: config.sh [hunk 4]
```diff
@@ -65,7 +74,7 @@ local_deployment_data="$(
 : "After assembling the configuration:"
 : "- replace OWN_CANISTER_ID"
 : "- construct ledger and governance canister URLs"
-jq -s '
+jq -s --sort-keys '
```
Comment (bitdivine) [inline config.sh:77]: Sorting entries by key makes spurious diffs from reordering entries disappear.

**Review by mstrasinskis — APPROVED** [review]
Rubber stamp. Thank you.

**Pull request merged**